python/paddle/fluid/tests/unittests/test_unsqueeze_op.py

```diff
@@ -0,0 +1,229 @@
+
+    def test_check_output(self):
+        self.check_output()
+
+    def test_check_grad(self):
+        self.check_grad(["X"], "Out")
+
+
+# Correct: Mixed input axis.
+class TestUnsqueezeOp2(OpTest):
+    def setUp(self):
+        ori_shape = (3, 5)
+        axes = (0, -1)
+        new_shape = (1, 3, 5, 1)
+
+        self.op_type = "unsqueeze"
+        self.inputs = {"X": np.random.random(ori_shape).astype("float32")}
+        self.attrs = {"axes": axes, "inplace": False}
+        self.outputs = {"Out": self.inputs["X"].reshape(new_shape)}
+
+    def test_check_output(self):
+        self.check_output()
+
+    def test_check_grad(self):
+        self.check_grad(["X"], "Out")
+
+
+# Correct: There is duplicated axis.
+class TestUnsqueezeOp3(OpTest):
+    def setUp(self):
+        ori_shape = (3, 2, 5)
+        axes = (0, 3, 3)
+        new_shape = (1, 3, 2, 1, 1, 5)
+
+        self.op_type = "unsqueeze"
+        self.inputs = {"X": np.random.random(ori_shape).astype("float32")}
+        self.attrs = {"axes": axes, "inplace": False}
+        self.outputs = {"Out": self.inputs["X"].reshape(new_shape)}
+
+    def test_check_output(self):
+        self.check_output()
+
+    def test_check_grad(self):
+        self.check_grad(["X"], "Out")
+
+
+# Correct: Inplace.
+class TestUnsqueezeOpInplace1(OpTest):
+    def setUp(self):
+        ori_shape = (3, 5)
+        axes = (0, 2)
+        new_shape = (1, 3, 1, 5)
+
+        self.op_type = "unsqueeze"
+        self.inputs = {"X": np.random.random(ori_shape).astype("float32")}
+        self.attrs = {"axes": axes, "inplace": True}
+        self.outputs = {"Out": self.inputs["X"].reshape(new_shape)}
+
+    def test_check_output(self):
+        self.check_output()
+
+    def test_check_grad(self):
+        self.check_grad(["X"], "Out")
+
+
+# Correct: Inplace. There is mins index.
+class TestUnsqueezeOpInplace2(OpTest):
+    def setUp(self):
+        ori_shape = (3, 5)
+        axes = (0, -2)
+        new_shape = (1, 3, 1, 5)
+
+        self.op_type = "unsqueeze"
+        self.inputs = {"X": np.random.random(ori_shape).astype("float32")}
+        self.attrs = {"axes": axes, "inplace": True}
+        self.outputs = {"Out": self.inputs["X"].reshape(new_shape)}
+
+    def test_check_output(self):
+        self.check_output()
+
+    def test_check_grad(self):
+        self.check_grad(["X"], "Out")
+
+
+# Correct: Inplace. There is duplicated axis.
+class TestUnsqueezeOpInplace3(OpTest):
+    def setUp(self):
+        ori_shape = (3, 2, 5)
+        axes = (0, 3, 3)
+        new_shape = (1, 3, 2, 1, 1, 5)
+
+        self.op_type = "unsqueeze"
+        self.inputs = {"X": np.random.random(ori_shape).astype("float32")}
+        self.attrs = {"axes": axes, "inplace": True}
+        self.outputs = {"Out": self.inputs["X"].reshape(new_shape)}
+
+    def test_check_output(self):
+        self.check_output()
+
+    def test_check_grad(self):
+        self.check_grad(["X"], "Out")
```

Comment:
unit tests can be a bit simpler by using inherit, like [here](https://github.com/PaddlePaddle/Paddle/blob/3d99e2e9df6333c52bca678c06c3ea51344b7894/python/paddle/fluid/tests/unittests/test_pool2d_op.py#L152-L164).

Reply:
get, thanks

Reply:
Done